File: backend/TrackYourBest/TYB.ApiService/Infrastructure/Entities/Spatial/TripStatus.cs
using NpgsqlTypes;

namespace TYB.ApiService.Infrastructure.Entities.Spatial
{
	public enum TripStatus
	{
		[PgName("driver_approve")]
		DriverApprove,
		[PgName("ongoing")]
		Ongoing,
		[PgName("paused")]
		Paused,
		[PgName("completed")]
		Completed,
		[PgName("cancelled_fm")]
		CancelledFm,
		[PgName("cancelled_driver")]
		CancelledDriver
	}
}

File: frontend/app/utils/tripStatus.ts
export function humanStatus(status: string | null | undefined): string {
  switch ((status ?? "").toLowerCase().replace(/[_\s]/g, "")) {
    case "driverapprove":   return "Awaiting Driver Approval";
    case "ongoing":         return "In Progress";
    case "paused":          return "Paused";
    case "completed":       return "Completed";
    case "cancelledfm":     return "Cancelled by Fleet Manager";
    case "cancelleddriver": return "Cancelled by Driver";
    default:                return "Unknown";
  }
}
